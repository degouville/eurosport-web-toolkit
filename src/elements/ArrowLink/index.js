import React from 'react';
import styled from 'react-emotion';
import PropTypes from 'prop-types';
import Chevron from 'src/assets/chevron.component.svg';
import Arrow from 'src/assets/right-arrow.component.svg';

const StyledChevron = styled(Chevron)`
  margin-left: 6px;

  path {
    fill: currentColor;
  }
`;

const StyledArrow = styled(Arrow)`
  margin-left: 6px;
  vertical-align: middle;
  path {
    fill: currentColor;
  }
`;

const LinkComponent = styled.a`
  color: ${({ theme }) => theme.link.color};
  &:focus,
  &:visited,
  &:hover {
    color: ${({ theme }) => theme.link.color};
  }
  display: flex;
  align-items: center;
  justify-content: center;
  text-decoration: none;
  text-align: center;
  text-transform: uppercase;
  font-size: 14px;
  font-weight: bold;
  letter-spacing: 1.27px;
  line-height: 24px;
  white-space: nowrap;
`;

const getArrow = arrowType => {
  switch (arrowType) {
    case 'chevron':
      return <StyledChevron />;
    case 'arrow':
      return <StyledArrow />;

    default:
      return null;
  }
};

const ArrowLink = ({ children, arrowType, ...props }) => (
  <LinkComponent {...props}>
    {children}
    {getArrow(arrowType)}
  </LinkComponent>
);

ArrowLink.displayName = 'ArrowLink';

ArrowLink.defaultProps = {
  arrowType: 'chevron',
};

ArrowLink.propTypes = {
  children: PropTypes.node.isRequired,
  arrowType: PropTypes.oneOf(['arrow', 'chevron', '']),
};

export default ArrowLink;
