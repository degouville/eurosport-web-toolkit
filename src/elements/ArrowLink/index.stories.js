import React from 'react';
import { storiesOf } from '@storybook/react';
import { text, radios } from '@storybook/addon-knobs';
import { ArrowLink } from '../..';

const indexStories = storiesOf('Elements|ArrowLink', module);

// eslint-disable-next-line import/prefer-default-export
export const arrowTypeOptions = {
  arrow: 'arrow',
  chevron: 'chevron',
  none: '',
};

indexStories.add('ArrowLink', () => (
  <ArrowLink href={text('href', 'goo.gl')} arrowType={radios('arrowType', arrowTypeOptions, 'chevron')}>
    {text('Label', 'Hello ArrowLink')}
  </ArrowLink>
));
