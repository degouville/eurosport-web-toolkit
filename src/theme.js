import { coreLightMinus1, featureOneMinus2, actionTwoDarkBase } from './colors';

export default {
  link: {
    color: featureOneMinus2,
  },
  button: {
    primary: {
      color: coreLightMinus1,
      backgroundColor: actionTwoDarkBase,
    },
    secondary: {
      color: featureOneMinus2,
    },
  },
};
