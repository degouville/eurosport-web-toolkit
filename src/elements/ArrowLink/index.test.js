import React from 'react';
import { shallow } from 'enzyme';
import theme from '../../theme';
import ArrowLink from '.';

it('renders a ArrowLink', () => {
  expect(
    shallow(
      <ArrowLink theme={theme} href="/">
        Click me !
      </ArrowLink>
    )
  ).toMatchSnapshot();
});
