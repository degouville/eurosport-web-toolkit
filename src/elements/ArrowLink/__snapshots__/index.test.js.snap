// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a ArrowLink 1`] = `
<Styled(a)
  href="/"
  theme={
    Object {
      "button": Object {
        "primary": Object {
          "backgroundColor": "#d50032",
          "color": "#ffffff",
        },
        "secondary": Object {
          "color": "#66cef9",
        },
      },
      "link": Object {
        "color": "#66cef9",
      },
    }
  }
>
  Click me !
  <Styled(ReactComponent) />
</Styled(a)>
`;
